pkgs/applications/networking/instant-messengers/signal-desktop/default.nix

```diff
@@ -3,9 +3,24 @@
 , dbus, libX11, xorg, libXi, libXcursor, libXdamage, libXrandr, libXcomposite
 , libXext, libXfixes, libXrender, libXtst, libXScrnSaver, nss, nspr, alsaLib
 , cups, expat, udev
+# Unfortunately this also overwrites the UI language (not just the spell
+# checking language!):
+, hunspellDicts, spellcheckerLanguage ? null # E.g. "de_DE"
+# For a full list of available languages:
+# $ cat pkgs/development/libraries/hunspell/dictionaries.nix | grep "dictFileName =" | awk '{ print $3 }'
 }:
 
 let
+  customLanguageWrapperArgs = (with lib;
+    let
+      # E.g. "de_DE" -> "de-de" (spellcheckerLanguage -> hunspellDict)
+      spellLangComponents = splitString "_" spellcheckerLanguage;
+      hunspellDict = elemAt spellLangComponents 0 + "-" + toLower (elemAt spellLangComponents 1);
+    in if spellcheckerLanguage != null
+      then ''
+        --set HUNSPELL_DICTIONARIES "${hunspellDicts.${hunspellDict}}/share/hunspell" \
+        --set LC_MESSAGES "${spellcheckerLanguage}"''
+      else "");
```

Comment:
You already merged it, but these parens are unnecessary

Reply:
@Infinisil Are you sure? You mean something like `--set LC_MESSAGES "$spellcheckerLanguage"''`, right? Normally that would work because Nix would create a corresponding shell variable for the attribute (IIRC this is done for all attributes passed to `mkDerivation`, if they're "representable in Bash"). But since the scope of e.g. `spellcheckerLanguage` is limited (by `let ... in ...`) the variable is never passed to `mkDerivation` and I therefore have to use the (Nix expression language) variable instead of the Bash variable. Or am I missing something?

Reply:
Oh, nvm, you've probably meant the parentheses around the if, yeah, I've missed that...